modules/core/04-channel/v2/keeper/grpc_query_test.go

```diff
@@ -587,6 +587,150 @@ func (suite *KeeperTestSuite) TestQueryNextSequenceSend() {
 	}
 }
 
+func (suite *KeeperTestSuite) TestQueryUnreceivedPackets() {
+	var (
+		expSeq []uint64
+		path   *ibctesting.Path
+		req    *types.QueryUnreceivedPacketsRequest
+	)
+
+	testCases := []struct {
+		msg      string
+		malleate func()
+		expPass  bool
```

Comment:
ah, final thing, we've gone with checking for specific errors instead of just pass/fail. If you take a look at other tests you'll see the `expError` pattern we use, should be relatively straight-forward to add.